engine/src/main/java/org/terasology/telemetry/TelemetryScreen.java

```diff
@@ -0,0 +1,177 @@
+        WidgetUtil.tryBindCheckBoxWithListener(this, "errorReportingEnabled",BindHelper.bindBeanProperty("errorReportingEnabled",config.getTelemetryConfig(),Boolean.TYPE), (checkbox) -> {
+            TelemetryLogstashAppender appender = TelemetryUtils.fetchTelemetryLogstashAppender();
+            if (config.getTelemetryConfig().isErrorReportingEnabled()) {
+                appender.turnOnErrorReporting();
+            } else {
+                appender.turnOffErrorReporting();
+            }
+        });
+    }
+
+    private Map<TelemetryCategory, Class> fetchTelemetryCategoriesFromEnvironment() {
+        Map<TelemetryCategory, Class> telemetryCategories = Maps.newHashMap();
+
+        DependencyResolver resolver = new DependencyResolver(moduleManager.getRegistry());
+        for (Name moduleId : moduleManager.getRegistry().getModuleIds()) {
+            Module module = moduleManager.getRegistry().getLatestModuleVersion(moduleId);
+            if (module.isCodeModule()) {
+                ResolutionResult result = resolver.resolve(moduleId);
+                if (result.isSuccess()) {
+                    try (ModuleEnvironment environment = moduleManager.loadEnvironment(result.getModules(), false)) {
+                        for (Class<?> holdingType : environment.getTypesAnnotatedWith(TelemetryCategory.class, new FromModule(environment, moduleId))) {
+                            TelemetryCategory telemetryCategory = holdingType.getAnnotation(TelemetryCategory.class);
+                            telemetryCategories.put(telemetryCategory, holdingType);
+                        }
+                    }
+                }
+            }
+        }
+
+        return telemetryCategories;
+    }
+
+    private void addTelemetrySection(TelemetryCategory telemetryCategory, ColumnLayout layout, Map<String,Object> map) {
+
+        UILabel categoryHeader = new UILabel(translationSystem.translate(telemetryCategory.displayName()));
+        categoryHeader.setFamily("subheading");
+        layout.addWidget(categoryHeader);
+        List<Map.Entry> telemetryFields = sortFields(map);
+        for (Map.Entry entry : telemetryFields) {
+            Object value = entry.getValue();
+            if (value instanceof List) {
+                List list = (List) value;
+                addTelemetryField(entry.getKey().toString(),list,layout);
+            }
+            else {
+                addTelemetryField(entry.getKey().toString(),value,layout);
+            }
+        }
+    }
+
+    private void addTelemetryField(String type, Object value, ColumnLayout layout) {
+        RowLayout newRow = new RowLayout(new UILabel(type),new UILabel(value.toString()))
+                .setColumnRatios(0.4f)
+                .setHorizontalSpacing(horizontalSpacing);
+
+        layout.addWidget(newRow);
+    }
+
+    // Will be used if the value is a List
```

Comment:
Even if this is a private method it might be worth using javadoc? That way I suspect an IDE would offer the comment when prompted (like `CTRL-Q` elsewhere in the class).

More of a general personal practice than a convention, really. I like javadoc, figure might as well use it :D